app/models/friend_request.rb

```diff
@@ -0,0 +1,31 @@
+class FriendRequest < ApplicationRecord
+  belongs_to :sender, class_name: 'User'
+  belongs_to :receiver, class_name: 'User'
+ 
+  # validate :not_friends
+  validate :not_pending
+  validate :not_self
+  validate :not_friends
+
+  def accept
```

Comment:
Great implementation here! I was going through, mentally, on the flow of how friendships will be created and then how the app will handle the associated friend_request. This answers that question in my mind! :) 

However, placing here the creation of the friendship may not be the best implementation. Later on when you do the `friendships_controller` the creation of friendship is better handled by the `create` action of that controller. That particular action will just need to invoke this method of your `friend_request` and delete the request accordingly. 

Another thing we need to consider here in the friend_request model is what happens if the receiver rejects the request (which happens lots of times on the real Facebook)? What I recommend is probably just simply calling the default `delete` method for `friend_request` for both `create` action in your `friendships_controller` and `destroy` action of your `friend_requests_controller`. The mechanisms of those will be:

1. When a receiver accepts the friend request -> Friendship record is created between the sender and the receiver -> Consequently, the friend_request record is deleted (action handled in the `friendships_controller`)

2. When a receiver rejects a friend_request -> friend_request record is deleted (action is handled in `friend_requests_controller`).